### PR TITLE
scenarios; MPIPolar boundary conditions; spherical test case; optional-dependencies in pyproject.toml; cleanups

### PR DESCRIPTION
**Pull request by Delcior**

### CODE REVIEW

File: tests/local/contract_tests/test_single_vs_multi_node.py [hunk 1]
```diff
@@ -133,59 +98,68 @@ def test_2d(
         plot_path = None
         if plot:
             plot_path = Path(os.environ["CI_PLOTS_PATH"]) / Path(
```
Comment (slayoo) [inline tests/local/contract_tests/test_single_vs_multi_node.py:100]: here we need to include scenario name to avoid overwriting files, right?

File: PyMPDATA_MPI/mpi_periodic.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# pylint: disable=invalid-name,unused-argument,c-extension-no-member,too-many-arguments
+# pylint: disable=invalid-name,unused-argument,c-extension-no-member,too-many-arguments,line-too-long,duplicate-code
```
Comment (slayoo) [inline PyMPDATA_MPI/mpi_periodic.py:1]: let's fix whatever is requireing `line-too-long` and `duplicate-code`!

File: PyMPDATA_MPI/mpi_periodic.py [hunk 3]
```diff
@@ -7,7 +7,12 @@
...
-from PyMPDATA.impl.enumerations import INVALID_INDEX, SIGN_LEFT, SIGN_RIGHT
+from PyMPDATA.impl.enumerations import SIGN_LEFT, SIGN_RIGHT
+
+from PyMPDATA_MPI.impl.boundary_condition_commons import (
+    make_scalar_boundary_condition,
+    make_vector_boundary_condition,
+)
 
 comm = MPI.COMM_WORLD
 IRRELEVANT = 666
```
Comment (slayoo) [inline PyMPDATA_MPI/mpi_periodic.py:18]: this seem not used anymore

File: README.md [hunk 4]
```diff
@@ -103,7 +103,7 @@ click D "https://pypi.org/p/PyMPDATA-MPI"
 click TESTS "https://pypi.org/p/PyMPDATA-MPI"
 ```
 Rectangular boxes indicate pip-installable Python packages (click to go to pypi.org package site).
-
+#TODO #80: REMOVE PBS FROM DIAG
```
Comment (slayoo) [inline README.md:106]: let's just remove it! :)

File: scenarios/spherical.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
+# pylint: disable=missing-module-docstring,missing-function-docstring,too-many-instance-attributes,too-few-public-methods
+# pylint: disable=missing-class-docstring,invalid-name,too-many-locals,too-many-arguments,c-extension-no-member, no-member
```
Comment (slayoo) [inline scenarios/spherical.py:2]: let's aim at removing all these pylint-disable comments

File: scenarios/cartesian.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
+# pylint: disable=missing-module-docstring,missing-function-docstring,missing-class-docstring,invalid-name,too-few-public-methods,too-many-locals
```
Comment (slayoo) [inline scenarios/cartesian.py:1]: let's aim at removing all these pylint-disable comments

File: tests/local/contract_tests/test_single_vs_multi_node.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-# pylint: disable=missing-module-docstring,missing-function-docstring,
+# pylint: disable=missing-module-docstring,missing-function-docstring,no-member
 # pylint: disable=missing-class-docstring,invalid-name,too-many-locals,too-many-arguments,c-extension-no-member
 # based on PyMPDATA README example
```
Comment (slayoo) [inline tests/local/contract_tests/test_single_vs_multi_node.py:3]: this comment is not valid anymore

File: .github/workflows/tests+pypi.yml [hunk 8]
```diff
@@ -131,7 +130,7 @@ jobs:
             export CI_PLOTS_PATH=plots/$arbitrarily_selected_artefact_set
```
Comment (slayoo) [inline .github/workflows/tests+pypi.yml:130]: does it need any new logic to handle multiple scenarios?

**Review by slayoo — CHANGES_REQUESTED** [review]

File: scenarios/spherical.py [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+                [self.settings.pdf_g_factor(i, j) for j in range(self.settings.nlat)]
+                for i in range(x0, x0 + mpi_nlon)
+            ]
+        )
+
+        # TODO #81: <move out>
+        Cx_max = np.amax(
+            np.abs((advector_x[1:, :] + advector_x[:-1, :]) / 2 / g_factor_z)
+        )
+        print(Cx_max)
```
Comment (slayoo) [inline scenarios/spherical.py:134]: debug leftover

File: scenarios/spherical.py [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+        assert Cx_max < 1
+
+        Cy_max = np.amax(
+            np.abs((advector_y[:, 1:] + advector_y[:, :-1]) / 2 / g_factor_z)
+        )
+        print(Cy_max)
```
Comment (slayoo) [inline scenarios/spherical.py:140]: debug leftover

**Review by slayoo — CHANGES_REQUESTED** [review]

File: PyMPDATA_MPI/mpi_periodic.py (hunk 3 above)
Comment (Delcior) [reply PyMPDATA_MPI/mpi_periodic.py]: removed

File: README.md [hunk 11]
```diff
@@ -103,7 +103,7 @@ click D "https://pypi.org/p/PyMPDATA-MPI"
 click TESTS "https://pypi.org/p/PyMPDATA-MPI"
 ```
 Rectangular boxes indicate pip-installable Python packages (click to go to pypi.org package site).
```
Comment (slayoo) [inline README.md:105]: BTW, we need to modify path for animations so that th Scenario component is featured

**Review by slayoo — APPROVED** [review]
